packages/api-db-mongodb/src/utility.ts

```diff
@@ -25,3 +25,7 @@ export function isNonNull<T>(value: T): value is NonNullable<T> {
 export enum MongoErrorCode {
   DuplicateKey = 11000
 }
+
+export function escapeRegExp(string: string) {
+  return string.replace(/[\-\[\]\/\{\}\(\)\*\+\?\.\\\^\$\|]/g, '\\$&')
```

Comment:
Why the double `\\` in the `\\$&`? I think it would be very good if you add a comment to explain what is happening here.